ext/etc/etc.c

```diff
@@ -226,17 +313,63 @@ etc_getpwuid(int argc, VALUE *argv, VALUE obj)
 #if defined(HAVE_GETPWENT)
     VALUE id;
     rb_uid_t uid;
-    struct passwd *pwd;
 
     if (rb_scan_args(argc, argv, "01", &id) == 1) {
 	uid = NUM2UIDT(id);
     }
     else {
 	uid = getuid();
     }
-    pwd = getpwuid(uid);
+
+# ifdef HAVE_GETPWUID_R
+    char *bufid;
```

Comment:
https://github.com/ruby/etc/actions/runs/10101599428/job/27935519316#step:5:40

> ```
> ../../../../ext/etc/etc.c:325:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
>   325 |     char *bufid;
>       |     ^~~~
> ```

Still the required ruby version is 2.6, we can't assume C99.